test/estimate-gas.test.js

```diff
@@ -0,0 +1,138 @@
+import { expect } from 'chai'
+const init = require('./helpers/init')
+const errors = require('./helpers/errors')
+const web3Utils = require('web3-utils')
+const ethjsABI = require('ethjs-abi')
+const createCsvWriter = require('csv-writer').createObjectCsvWriter
+
+
+contract('Gas Estimates', accounts => {
+    const ctx = {
+      actors: init.getTestActorsContext(accounts),
+      contracts: {
+        token: null,
+        atonomi: null
+      }
+    }
+
+    const csvWriter = createCsvWriter({
+        path: 'output/gasOutput.csv',
+        header: [
+            {id: 'scope', title: 'SCOPE'},
+            {id: 'task', title: 'TASK'},
+            {id: 'gas', title: 'COST IN WEI'},
+        ]
+    })
+    const recordsForCSV = []
+    const tokenDecimals = 18
+    const multiplier = 10 ** tokenDecimals
+    const regFee = 1 * multiplier
+    const actFee = 1 * multiplier
+    const repReward = 1 * multiplier
+    const irnReward = repReward * 0.20
+    const mfgReward = repReward - irnReward
+    const deviceId = 'apple-iphone1'
+    const deviceIdHash = web3Utils.soliditySha3({t: 'bytes32', v: web3.fromAscii(deviceId)})
+    const mfgId = 'APPLE'
+    const deviceType = 'phone'
+    const devicePublicKey = 'somepublickey'
+    const score = 'somescore'
+
+    beforeEach(async () => {
+        ctx.contracts.token = await init.getAtonomiTokenContract(ctx.actors.owner, ctx.actors.releaseAgent)
+        ctx.contracts.atonomi = await init.getAtonomiContract(ctx.actors.owner, ctx.contracts.token.address)
+        await ctx.contracts.token.transfer(ctx.actors.deviceOwner, actFee, {from: ctx.actors.owner})
+        await ctx.contracts.token.transfer(ctx.actors.mfg, regFee + actFee, {from: ctx.actors.owner})
+        await ctx.contracts.atonomi.addNetworkMember(ctx.actors.mfg, false, true, false, mfgId, {from: ctx.actors.owner})
+        await ctx.contracts.atonomi.addNetworkMember(ctx.actors.irnNode, false, false, true, '', {from: ctx.actors.owner})
+    })
+
+    describe('global costs', () => {
+        it('   ^= to Deposit Tokens', async() => {
+            var gas = await ctx.contracts.token.transfer.estimateGas(ctx.actors.alice, regFee, {from: ctx.actors.owner})
+            console.log('      ✓ ' + gas + ' wei')
+            recordsForCSV.push( { scope: 'global', task: 'deposit tokens', gas: gas } )
+        })
+    })
+    
+    describe('atonomi costs', () => {
+        it('   ^= to Add Admin', async () => {
+            var gas = await ctx.contracts.atonomi.addNetworkMember.estimateGas(ctx.actors.admin, true, false, false, '', {from: ctx.actors.owner})
+            console.log('      ✓ ' + gas + ' wei')
+            recordsForCSV.push( { scope: 'atonomi', task: 'add admin', gas: gas } )
+        })
+        it('   ^= to Add Manufacturer', async () => {
+            var gas = await ctx.contracts.atonomi.addNetworkMember.estimateGas(ctx.actors.admin, false, true, false, '', {from: ctx.actors.owner})
```

Comment:
update actor to be `ctx.actors.mfg`, also it should supply a manufacturer id (not empty string).  I'm surprised it didnt throw though, can you investigate a little further?  See the add mfg tests again.

Reply:
currently, we allow for a null memberId, and replace it with ' ':
```javascript
48    expect(memberId).to.be.equal(web3.toAscii(log.args._memberId).replace(/\u0000/g, ''))
```
(from [network-management.test.js](https://github.com/atonomi/smart-contracts/blob/master/test/network-management.test.js))

Reply:
But for manufacturers it should not be null.  We have validation in there to enforce that in the contract

Reply:
added issue #79 to address this